src/Workspaces/Core/Portable/SolutionCrawler/SolutionCrawlerOptions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.SolutionCrawler
+{
+    internal static class SolutionCrawlerOptions
+    {
+        /// <summary>
+        /// Option to turn configure background analysis scope.
+        /// </summary>
+        public static readonly Option<BackgroundAnalysisScope> BackgroundAnalysisScopeOption = new Option<BackgroundAnalysisScope>(
+            nameof(SolutionCrawlerOptions), nameof(BackgroundAnalysisScopeOption), defaultValue: BackgroundAnalysisScope.Default,
+            storageLocations: new RoamingProfileStorageLocation($"TextEditor.BackgroundAnalysisScopeOption"));
+
+        /// <summary>
+        /// Enables forced <see cref="BackgroundAnalysisScope.Minimal"/> scope when low VM is detected to improve performance.
+        /// </summary>
+        public static bool LowMemoryForcedMinimalBackgroundAnalysis = false;
+
+        public static BackgroundAnalysisScope GetBackgroundAnalysisScope(Project project)
+            => GetBackgroundAnalysisScope(project.Solution.Options);
+
+        public static BackgroundAnalysisScope GetBackgroundAnalysisScope(OptionSet options)
+        {
+            if (LowMemoryForcedMinimalBackgroundAnalysis)
+            {
+                return BackgroundAnalysisScope.Minimal;
+            }
+
+            return options.GetOption(BackgroundAnalysisScopeOption);
```

Comment:
i don't get the relationshi between GetOverriddenBackgroundAnalysisScope and this method.

Reply:
`GetOverriddenBackgroundAnalysisScope` is just a mechanism to allow each individual incremental analyzer to have it's own option for BG analysis scope. For example, test explorer needs this override. `BackgroundAnalysisScopeOption` is the core option applicable to rest of the incremental analyzers that do not have a special BG analysis scope option.